sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -52,9 +52,30 @@
 
     @Override
     public Mono<HttpResponse> send(HttpRequest request) {
+        return send(request, Context.NONE);
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        boolean eagerlyReadResponse = context.getData("eagerly-read-response")
+            .map(data -> Boolean.parseBoolean(data.toString()))
+            .orElse(false);
```

Comment:
This seems like an unfortunate perf regression. Why wouldn't the value be stored as a Boolean in the Context?

Reply:
Updated